test/fetch_multi_by_test.rb

```diff
@@ -0,0 +1,61 @@
+require "test_helper"
+
+class FetchMultiByTest < IdentityCache::TestCase
+  NAMESPACE = IdentityCache::CacheKeyGeneration::DEFAULT_NAMESPACE
+
+  def setup
+    super
+    @bob = Item.new
+    @bob.id = 1
+    @bob.title = 'bob'
+
+    @bertha = Item.new
+    @bertha.id = 2
+    @bertha.title = 'bertha'
+  end
+
+  def test_fetch_multi_by_cache_key
+    Item.cache_index :title, unique: false
+
+    @bob.save!
+    @bertha.save!
+
+    assert_equal [@bob], Item.fetch_by_title('bob')
+
+    assert_equal [@bob, @bertha], Item.fetch_multi_by_title(['bob', 'bertha'])
```

Comment:
I wonder if it would be more consistent to return `[[@bob], [@bertha]]` here.

Reply:
`fetch_multi_by_title` would always have a predictable return type.  Wrapping each value in an array seems unnecessarily inconvenient from the perspective of the caller of that method.  Consistency would only make sense if the caller was something generic that had to work with any of these indexes, since that seems like the only type of caller would have to special case single column return values.

Also, this behaviour is consistent with how active record's `pluck` method works, which similarly asks for a fixed set of columns.

Reply:
oh wait, this is supposed to be a non-unique index.  I guess the problem is that the caller would have to associate the given values back to the given arguments.

Reply:
`fetch_multi_by_#{key}` has the same return type whether the cache is `unique: true` or not.

All other fetch methods, including `fetch_multi_#{alias}_by_#{key}`, have _different_ return types depending on the uniqueness of the cache.

Reply:
If we use nested arrays for `fetch_multi_by_#{key}`, then it seems like that would imply that it would always return an array of the same size as the given keys, such that it could be be used with `zip`ed together with the input keys to get a hash of results (e.g. `Hash[Item.fetch_multi_by_title(titles).zip(titles)]`).

However, I worry that might set the expectation that `fetch_multi_by_#{key}` and `fetch_multi` will work similarly.  However, `fetch_multi` returns a compacted array, which means it wouldn't work properly with `zip` using the input keys array.